backends/dpdk/dpdkHelpers.cpp

```diff
@@ -159,10 +159,42 @@ bool ConvertStatementToDpdk::preorder(const IR::AssignmentStatement *a) {
         auto mi = P4::MethodInstance::resolve(m, refmap, typemap);
         if (auto e = mi->to<P4::ExternMethod>()) {
             if (e->originalExternType->getName().name == "Hash") {
+                auto di = e->object->to<IR::Declaration_Instance>();
+                auto declArgs = di->arguments;
+                auto hash_alg = declArgs->at(0)->expression;
+                unsigned hashAlgValue = 0;
+                if (hash_alg->is<IR::Constant>())
+                    hashAlgValue = hash_alg->to<IR::Constant>()->asUnsigned();
+                cstring hashAlgName = "";
+                if (hashAlgValue == JHASH0 || hashAlgValue == JHASH5)
+                    hashAlgName = "jhash";
+                else if (hashAlgValue >= CRC1 && hashAlgValue <= CRC4)
+                    hashAlgName = "crc32";
+
+                IR::Vector<IR::Expression> components;
```

Comment:
Limit the scope for "components", move the declaration to if block.

Reply:
not addressed?

Reply:
"components" is now used in both if and else if part. So defining it outside if block